core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/preferences/TitleBlockPreferencePage.java

```diff
@@ -255,19 +255,12 @@ public boolean performOk() {
     for (int i = 0; i < linesNumber; i++) {
       for (int j = 0; j < columnsNumber; j++) {
         if (i == linesNumber - 1 && j == columnsNumber - 1) {
-          if (tccMatrix.get(i).get(j).name.equals(EMPTY_STRING)
-              || tccMatrix.get(i).get(j).content.equals(EMPTY_STRING)) {
-            createMessageBox(EDIT_ALL_CELLS, ERROR);
-            return false;
+          if (tccMatrix.get(i).get(j).content.equals(EMPTY_STRING)) {
+        	  tccMatrix.get(i).get(j).content = " ";
           }
           currentTableContent = currentTableContent + tccMatrix.get(i).get(j).name + SEPARATOR
               + tccMatrix.get(i).get(j).content;
-        } else {
-          if (tccMatrix.get(i).get(j).name.equals(EMPTY_STRING)
-              || tccMatrix.get(i).get(j).content.equals(EMPTY_STRING)) {
-            createMessageBox(EDIT_ALL_CELLS, ERROR);
-            return false;
-          }
+        } else {             	 

```

Comment:
The user can add a line or a column at any position, not only at the end.
Why do we only check the last cell ?

Reply:
If the last cell contains an empty content, when string is build, it will appear something like this: String s = "...SEPARATORSEPARATOR". When perform split on this string, it will create an array that will not contain the empty content of the last cell (and instead of 12 elements for example, it will contain only 11). Mostly for this is that check. Maybe it could be removed that else and add SEPARATOR after every content/name, even if is the last cell, I'm not sure about this :-?